### PR TITLE
DOC: Update contributing.rst to give suggestions on non-code contributions

### PR DESCRIPTION
**Pull request by logankilpatrick**
Based on a conversation at the NumFOCUS project summit.

### CODE REVIEW

**Comment by logankilpatrick** [comment]
I am not an RST expert so apologies if there are any weird mistakes here

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -13,6 +13,33 @@ Contributing to pandas
...
 enhancements, and ideas are welcome.
 
+.. _contributing.non_code:
+
+Non-Code Contributions
+======================
+
+Apart from code contributions, we value and welcome other forms of contributions to help grow and foster a supportive community. Here are some ways you can contribute without writing a single line of code:
+
+- **Documentation Improvements**: Help us improve our documentation by fixing typos, adding examples, or clarifying sections. You can find guidelines for contributing to documentation `here <https://pandas.pydata.org/docs/development/contributing_documentation.html>`_.
```
Comment (WillAyd) [inline doc/source/development/contributing.rst:23]:
```suggestion
- **Documentation Improvements**: Help us improve our documentation by fixing typos, adding examples, or clarifying sections. See also our documentation on `contributing to documentation <https://pandas.pydata.org/docs/development/contributing_documentation.html>`_.

I don't think sphinx/rst likes using the name `here` for different locations; doing something like this should help you work around that limitation
```

**Review by WillAyd — COMMENTED** [review]
@noatamir thoughts on this?

**Review by datapythonista — COMMENTED** [review]
Thanks @logankilpatrick for the contribution.

Do you mind limiting the line length to something like 80 or 100 please? We don't have a strong limiting, but very long lines are more difficult to deal with, for example in git diff, so we try to keep our docs this way.

For the list, can you remove the blank lines between list items? I think Sphinx won't render the list correctly otherwise.

Thank you!

**Comment by datapythonista** [comment]
Also, this CI problems seem to be caused by your changes, if you can have a look: https://github.com/pandas-dev/pandas/actions/runs/6157790957/job/16709259450?pr=55107#step:6:65

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**